go/oasis-node/cmd/config/migrate/migrate.go

```diff
@@ -0,0 +1,617 @@
+								m(m(newCfg["consensus"])["debug"])["disable_addr_book_from_genesis"] = dabfg
+								delete(m(m(m(tendermint)["seed"])["debug"]), "disable_addr_book_from_genesis")
+							}
+						}
+					} else if k == "sentry" {
+						if ua, ok := m(v)["upstream_address"]; ok {
+							logger.Info("consensus.tendermint.sentry.upstream_address is now consensus.sentry_upstream_addresses")
+							m(newCfg["consensus"])["sentry_upstream_addresses"] = ua
+							delete(m(m(tendermint)["sentry"]), "upstream_address")
+						}
+					} else if k == "upgrade" {
+						if sd, ok := m(v)["stop_delay"]; ok {
+							logger.Info("consensus.tendermint.upgrade.stop_delay is now consensus.upgrade_stop_delay")
+							m(newCfg["consensus"])["upgrade_stop_delay"] = sd
+							delete(m(m(tendermint)["upgrade"]), "stop_delay")
+						}
+					} else if k == "supplementarysanity" {
+						logger.Info("consensus.tendermint.supplementarysanity.* is now consensus.supplementary_sanity.*")
+						m(newCfg["consensus"])["supplementary_sanity"] = v
+						delete(m(tendermint), k)
+					} else if k == "p2p" {
+						mkSubMap(m(newCfg["consensus"]), "p2p")
+						for pk, pv := range m(v) {
+							nk := pk
+							if pk == "persistent_peer" {
+								logger.Info("consensus.tendermint.p2p.persistent_peer is now consensus.p2p.persistent_peers")
+								nk = "persistent_peers"
+							} else if pk == "unconditional_peer" {
+								logger.Info("consensus.tendermint.p2p.unconditional_peer is now consensus.p2p.unconditional_peers")
+								nk = "unconditional_peers"
+							} else if pk == "seed" {
+								logger.Info("consensus.tendermint.p2p.seed is now p2p.seeds")
+								mkSubMap(newCfg, "p2p")
+								m(newCfg["p2p"])["seeds"] = pv
+								delete(m(m(tendermint)["p2p"]), pk)
+								continue
+							}
+							m(m(newCfg["consensus"])["p2p"])[nk] = pv
+							delete(m(m(tendermint)["p2p"]), pk)
+						}
+					} else if k == "abci" {
+						if prune, ok := m(v)["prune"]; ok {
+							logger.Info("consensus.tendermint.abci.prune.* is now consensus.prune.*")
+							m(newCfg["consensus"])["prune"] = prune
+							delete(m(m(tendermint)["abci"]), "prune")
+						}
+					} else {
+						m(newCfg["consensus"])[k] = v
+						delete(m(tendermint), k)
+					}
+				}
+			}
+		}
+	}
+	if nodeMode != "client" {
+		newCfg["mode"] = nodeMode
+	}
+
+	runtime, ok := oldCfg["runtime"]
+	if ok {
+		if mode, ok := m(runtime)["mode"]; ok {
+			logger.Warn("runtime.mode has been deprecated in favor of using the global node mode")
+			if mode != "none" && mode != nodeMode {
+				logger.Warn("node mode set to runtime.mode", "mode", mode)
+				newCfg["mode"] = mode
+			}
+			delete(m(runtime), "mode")
+		}
+
+		if sandbox, ok := m(runtime)["sandbox"]; ok {
+			if binary, ok := m(sandbox)["binary"]; ok {
+				logger.Info("runtime.sandbox.binary is now runtime.sandbox_binary")
+				mkSubMap(newCfg, "runtime")
+				m(newCfg["runtime"])["sandbox_binary"] = binary
+				delete(m(runtime), "sandbox")
```

Comment:
Yes, these sections only have one value and after the migration a pruning step is done which deletes empty sections anyways, so deleting only the value here would have the same effect on a valid config file.

I will change it to delete just the value in case anyone has invalid options in the same section of their config file.